src/com/lftechnology/java/training/srijan/basics/coreapp/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,305 @@
+				LOGGER.log(Level.INFO, "Failed to connect to database");
+			}
+		} catch (SQLException e1) {
+			LOGGER.log(Level.INFO, "SQl Exception{0}", e1);
+		} finally {
+			try {
+				resultSet.close();
+				if (prepareStatementLoginCheck != null) {
+					prepareStatementLoginCheck.close();
+				}
+				if (con != null) {
+					con.close();
+				}
+			} catch (SQLException e1) {
+				LOGGER.log(Level.INFO, "SQLException{0}", e1);
+			}
+
+		}
+		return UserRole.INVALID;
+	}
+
+	@Override
+	public List<Employee> viewEmployeeDetails() {
+		Statement statement = null;
+		ResultSet resultSet = null;
+		Connection con = null;
+		List<Employee> employees = new ArrayList<>();
+		con = DbConnection.dbConnection();
+		if (con != null) {
+			System.out.println("inside view employeedetails");
+			String query = "Select * from employeeDetails";
+			try {
+				statement = con.createStatement();
+				resultSet = statement.executeQuery(query);
+				while (resultSet.next()) {
+					Employee employee = new Employee();
+					employee.setId(resultSet.getInt("id"));
+					employee.setUserName(resultSet.getString("username"));
+					employee.setFullName(resultSet.getString("fullName"));
+					employee.setDepartment(resultSet.getString("department"));
+					employee.setAddress(resultSet.getString("address"));
+					employee.setIsTerminated(resultSet.getBoolean("is_terminated"));
+					employees.add(employee);
+				}
+			} catch (SQLException e1) {
+				LOGGER.log(Level.INFO, "SQL Exception{0}", e1);
+			} finally {
+				try {
+					statement.close();
+					con.close();
+				} catch (SQLException e1) {
+					LOGGER.log(Level.INFO, "SQL Exception{0}", e1);
+				}
+			}
+
+		}
+		return employees;
+	}
+
+	@Override
+	public void terminate(int idToTerminate) {
+		Connection con = null;
+		ResultSet resultSet = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareStatementUpdateInfo = null;
+		PreparedStatement prepareUpdate = null;
+		try {
+			if (con != null) {
+				String query = "Select * from employeeDetails where id = ?";
+				prepareStatementUpdateInfo = con.prepareStatement(query);
+				prepareStatementUpdateInfo.setInt(1, idToTerminate);
+				resultSet = prepareStatementUpdateInfo.executeQuery();
+				while (resultSet.next()) {
+					Boolean status = resultSet.getBoolean("is_terminated");
+					status = !status;
+					String queryUpdate = "UPDATE employeeDetails SET is_terminated = ? where id = ? ";
+					prepareUpdate = con.prepareStatement(queryUpdate);
+					prepareUpdate.setBoolean(1, status);
+					prepareUpdate.setInt(2, idToTerminate);
+					prepareUpdate.executeUpdate();
+				}
+			}
+		} catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception:{0}", ex);
+		} finally {
+			try {
+				resultSet.close();
+				prepareStatementUpdateInfo.close();
+				prepareUpdate.close();
+				con.close();
+			} catch (SQLException ex) {
+				LOGGER.log(Level.INFO, "SQL Exception {0}", ex);
+			}
+		}
+	}
+
+	public void updateEmployee(Employee e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)